pallets/teerex/src/lib.rs

```diff
@@ -355,8 +376,18 @@ pub mod pallet {
 			mr_enclave: MREnclave,
 		) -> DispatchResultWithPostInfo {
 			T::EnclaveAdminOrigin::ensure_origin(origin)?;
+			ensure!(
+				!ScheduledEnclaveIndex::<T>::contains_key(mr_enclave),
+				Error::<T>::ScheduledEnclaveAlreadyExist
+			);
 			ScheduledEnclave::<T>::insert(sidechain_block_number, mr_enclave);
 			Self::deposit_event(Event::UpdatedScheduledEnclave(sidechain_block_number, mr_enclave));
+			let enclave_idx = Self::scheduled_enclave_count()
+				.checked_add(1)
+				.ok_or("[Teerex]: Overflow adding new MrEnclave to registry")?;
+			<ScheduledEnclaveRegistry<T>>::insert(enclave_idx, mr_enclave);
+			<ScheduledEnclaveIndex<T>>::insert(mr_enclave, enclave_idx);
+			<ScheduledEnclaveCount<T>>::put(enclave_idx);
```

Comment:
As the enclave_idx is always increasing, we might have a problem when we want to override an existing schedule, like `update_scheduled_enclave(10, MR1)`, then `update_scheduled_enclave(10, MR2)`. It's not so uncommon to do so